### PR TITLE
completed assignment 52

### PR DESCRIPTION
**Pull request by npauzenga**
@Enriikke so I was really into our talk about testing the other night and wanted to jump ahead a little. In the test I've submitted I'm not actually sure if I can use logical operators in the `eq()` method but we're going to find out :grinning: 

In other news, I've got Travis and Coveralls setup and working (though obviously everything's failing). Cool stuff! Mind checking over my installs? I think there's probably some more work to be done in the `spec_helper` and `.travis.yml`...

### CODE REVIEW

File: spec/models/vote_spec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require "rails_helper"
+
+describe Vote do
+  describe "validations" do
+    describe "value validation" do
+      it "only allows -1 or 1 as values" do
+        expect(@vote.value).to eq(true || false)
```
Comment (Enriikke) [inline spec/models/vote_spec.rb]: In this case, we want to test that a `Vote`'s `value` can only be initialized to **1** or **-1**. This means that after initializing a `Vote` object with a different number, if you try to `save` it, would return false.

`eq` should only be expecting one thing, depending of what you are testing. For example, if you initialized a `Vote` object with the value **1**, then your expectation should be that saving the vote `eq(true)`. On the other hand, if you initialize a `Vote` object with the value **2**, then your expectation should be `eq(false)`.

**Comment by Enriikke** [comment]
@npauzenga your spec is not testing what you should right now. I read through the instructions and I can see where the confusion was :stuck_out_tongue: 

Think about the flow, you are creating (**initializing**) `Vote` objects, and you want to make sure that they can only be saved if the `value` attribute is either **1** or **-1**. Remember that an `ActiveRecord` model has a `save` method, which returns `true` if it was successfully saved, or `false` otherwise.

I also noticed that you commented out most of the `spec_helper`. There are some good default configurations that might be beneficial. We can talk through what those mean during our session. I will suggest you add the configurations back and we can walk through any errors you might be getting :smiley: 

Awesome work setting up travis and coveralls!!! :boom: :sunglasses: I'm sure you will find them beneficial as your test suite grows and specially once you move to the projects!!

**Comment by npauzenga** [comment]
@Enriikke aha! So I actually tackled this assignment before learning a little bit about validations. That'll teach me to go out of order :smile: 

On the `spec_helper` -- I believe it was actually generated with those settings commented out. They'd used a block comment (which rubocop doesn't like) so I changed them to line comments. On this commit I've uncommented those lines. Please disregard Hound's barking. If we could go through some of these options I'll clean up the file a bit. Just wanted to keep all of the comments there in the meantime.

**Comment by npauzenga** [comment]
As a follow-up it's this line -- `config.disable_monkey_patching!` that breaks things. Perhaps we could talk a bit about it...

File: spec/models/vote_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require "rails_helper"
+
+describe Vote do
+  describe "validations" do
+    describe "value validation" do
+      it "only allows -1 or 1 as values" do
+        expect(@vote.save).to eq(true)
```
Comment (Enriikke) [inline spec/models/vote_spec.rb]: I think here we are missing the declaration of `@vote`. I remember seeing it in our previous chat but it looks like maybe you forgot to check that in :smile:.

**Comment by npauzenga** [comment]
@Enriikke yep! The changes we talked about yesterday are actually included in the next checkpoint's submits (coming soon). I'll go back this evening and include 'em here as well.

**Comment by npauzenga** [comment]
There we go. Incorporated the changes we talked about yesterday.

**Comment by Enriikke** [comment]
Awesome work @npauzenga!!! :shipit:

**Pull request merged**